src/PrestaShopBundle/Form/Admin/Improve/International/Locations/CountryType.php

```diff
@@ -0,0 +1,189 @@
+use PrestaShopBundle\Form\Admin\Type\ShopChoiceTreeType;
+use PrestaShopBundle\Form\Admin\Type\SwitchType;
+use PrestaShopBundle\Form\Admin\Type\TranslatableType;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+use Symfony\Component\Validator\Constraints\Length;
+
+class CountryType extends AbstractType
+{
+    /**
+     * @var TranslatorInterface
+     */
+    protected $translator;
+
+    /**
+     * @var bool
+     */
+    protected $isMultistoreEnabled;
+
+    /**
+     * @var FormChoiceProviderInterface
+     */
+    protected $currencyChoiceProvider;
+
+    /**
+     * @var ConfigurableFormChoiceProviderInterface
+     */
+    protected $zoneChoiceProvider;
+
+    /**
+     * ZoneType constructor.
+     */
+    public function __construct(
+        TranslatorInterface $translator,
+        bool $isMultistoreEnabled,
+        FormChoiceProviderInterface $currencyChoiceProvider,
+        ConfigurableFormChoiceProviderInterface $zoneChoiceProvider
+    ) {
+        $this->translator = $translator;
+        $this->isMultistoreEnabled = $isMultistoreEnabled;
+        $this->currencyChoiceProvider = $currencyChoiceProvider;
+        $this->zoneChoiceProvider = $zoneChoiceProvider;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add('name', TranslatableType::class, [
+                'label' => $this->translator->trans('Country', [], 'Admin.Global'),
+                'options' => [
+                    'constraints' => [
+                        new Length([
+                            'max' => 64,
+                            'maxMessage' => $this->translator->trans(
+                                'This field cannot be longer than %limit% characters.',
+                                ['%limit%' => 64],
+                                'Admin.Notifications.Error'
+                            ),
+                        ]),
+                        new TypedRegex([
+                            'type' => 'generic_name',
+                        ]),
+                    ],
+                ],
+                'constraints' => [
+                    new DefaultLanguage(),
+                ],
+            ])
+            ->add('iso_code', TextType::class, [
+                'required' => true,
+                'label' => $this->translator->trans('ISO code', [], 'Admin.International.Feature'),
+                'constraints' => [
+                    new Length([
+                        'max' => 2,
+                        'maxMessage' => $this->translator->trans(
+                            'This field cannot be longer than %limit% characters.',
+                            ['%limit%' => 2],
```

Comment:
I see you already use a TypedRegex constraint bellow which already ensures the pattern is correct.
I was going to suggest not using Length constraint at all (because TypedRegex already takes care of that), but as I see the regex pattern actually allows 2 to 3 chars, while this length constraint only allows 2. So Im pretty sure we should fix the pattern in IsoCode VO, wdyt @jolelievre ? (as comment in IsoCode VO it `Stores language's two-letter (639-1) ISO code`). Furthermore if we still want to use the Length constraint for clearer error message, we should add the length constant to IsoCode VO and use it here instead of these magic numbers.

Reply:
You're right the `Length` constraint is redundant here so it can be removed.

Regarding the ISO code regexp itself it's the same regexp that was used in the `Validate` legacy class for years so it might be risky to change it in this PR. When I looked at ISO code definition from Wikipedia https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes I can see that it can be two or three letters depending on ISO norm used ISO 639-1 (two letters) or 639-2 and 639-3 (three letters).

But it's not like one norm accepts both so we should settle for a defined norm, seems like we are using the ISO 639-1 since all our default fixtures use two letters, but anyway it's not the scope of this PR this should be investigated separately.